File: client/src/app/modules/tiers/services/tiers.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { TiersType } from '../../tiers/model/TiersType';

@Injectable({
  providedIn: 'root'
})
export class TiersTypeService extends AppRestService<TiersType>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getTiersTypes() {
    return this.getListCached(new HttpParams(), "tiers-types");
  }

  addOrUpdateTiersType(tiersType: TiersType) {
    this.clearListCache(new HttpParams(), "tiers-types");
    return this.addOrUpdate(new HttpParams(), "tiers-type", tiersType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/tiers/service/TiersTypeService.java
package com.jss.osiris.modules.osiris.tiers.service;

import java.util.List;

import com.jss.osiris.modules.osiris.tiers.model.TiersType;

public interface TiersTypeService {
    public List<TiersType> getTiersTypes();

    public TiersType getTiersType(Integer id);

    public TiersType addOrUpdateTiersType(TiersType tiersType);
}
